src/foam/nanos/crunch/PredicatedPrerequisiteCapabilityJunctionDAO.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'PredicatedPrerequisiteCapabilityJunctionDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: `
+    A DAO decorator which filters prerequisitejunctions returned by find/select by the predicate
+    set on the junction.
+  `,
+
+  javaImports: [
+    'foam.mlang.predicate.AbstractPredicate',
+    'foam.nanos.auth.AuthService',
+    'static foam.mlang.MLang.AND'
+  ],
+  
+  constants: [
+    {
+      type: 'String',
+      name: 'PERMISSION',
+      value: 'predicatedprequisite.read.*'
```

Comment:
```suggestion
      value: 'predicatedprerequisite.read.*'
```
I sometimes make that typo too. There's a few other places in this pr with it.